group-coordinator/src/test/java/org/apache/kafka/coordinator/group/runtime/CoordinatorRuntimeTest.java

```diff
@@ -950,6 +948,55 @@ public void testScheduleWriteOpWhenWriteFails() {
         assertEquals(mkSet("record1", "record2"), ctx.coordinator.coordinator().records());
     }
 
+    @Test
+    public void testScheduleWriteOpWhenWriteTimesout() throws InterruptedException {
+        MockTimer timer = new MockTimer();
+        MockTime time = new MockTime();
+        // The partition writer only accept on write.
+        MockPartitionWriter writer = mock(MockPartitionWriter.class);
+
+        CoordinatorRuntime<MockCoordinatorShard, String> runtime =
+            new CoordinatorRuntime.Builder<MockCoordinatorShard, String>()
+                .withTime(timer.time())
+                .withTimer(timer)
+                .withLoader(new MockCoordinatorLoader())
+                .withEventProcessor(new DirectEventProcessor())
+                .withPartitionWriter(writer)
+                .withCoordinatorShardBuilderSupplier(new MockCoordinatorShardBuilderSupplier())
+                .withCoordinatorRuntimeMetrics(mock(GroupCoordinatorRuntimeMetrics.class))
+                .withCoordinatorMetrics(mock(GroupCoordinatorMetrics.class))
+                .build();
+
+        // Loads the coordinator.
+        runtime.scheduleLoadOperation(TP, 10);
+
+        // Verify the initial state.
+        CoordinatorRuntime<MockCoordinatorShard, String>.CoordinatorContext ctx = runtime.contextOrThrow(TP);
+        assertEquals(0, ctx.lastWrittenOffset);
+        assertEquals(0, ctx.lastCommittedOffset);
+        assertEquals(Collections.singletonList(0L), ctx.snapshotRegistry.epochsList());
+
+        // The partition writer times out when trying to append records.
+        doAnswer(invocation -> {
+            time.sleep(DEFAULT_WRITE_TIMEOUT * 3);
```

Comment:
You actually don't need this because the timeout will be reached if the HWM is not advanced.